src/utils/fetchClient.js

```diff
@@ -0,0 +1,30 @@
+const BASE_URL = 'https://mate.academy/students-api';
+
+
+function wait(delay) {
+  return new Promise(resolve => {
+    setTimeout(resolve, delay);
+  });
+}
+
+function request(url, method = 'GET', data = null) {
+  const options = { 
+    method,
+    body: data ? JSON.stringify(data) : undefined,
+    headers: data 
+    ? { 'Content-Type': 'application/json; charset=utf-8'}
+    : undefined
+   };
+
+    return wait(300)
+    .then(() => fetch(BASE_URL + url, options))
+    .then(response => response.json());
```

Comment:
Consider adding error handling for the fetch request to manage network errors or unsuccessful HTTP responses. This will make the client more robust.